tests/src/kakarot/accounts/test_account_contract.py

```diff
@@ -351,176 +348,508 @@ def test_should_execute_starknet_call(self, cairo_run):
+                    },
+                ),
+                "chainId": CHAIN_ID,
+            }
+            encoded_unsigned_tx = rlp_encode_signed_data(transaction)
+            tx_data = list(encoded_unsigned_tx)
             private_key = generate_random_private_key()
             address = int(private_key.public_key.to_checksum_address(), 16)
             signed = Account.sign_transaction(transaction, private_key)
+            signature = [
+                *int_to_uint256(signed.r),
+                *int_to_uint256(signed.s),
+                signed["v"],
+            ]
+
+            with cairo_error(message="Invalid chain id"), SyscallHandler.patch(
+                "Account_evm_address", address
+            ):
+                cairo_run(
+                    "test__execute_from_outside",
+                    tx_data=tx_data,
+                    signature=signature,
+                    chain_id=CHAIN_ID + 1,
+                )
 
+        @SyscallHandler.patch(
+            "IKakarot.get_cairo1_helpers_class_hash",
+            lambda addr, data: [CAIRO1_HELPERS_CLASS_HASH],
+        )
+        @SyscallHandler.patch(
+            "Account_nonce",
+            100,
+        )
+        @pytest.mark.parametrize("transaction", TRANSACTIONS)
+        def test_should_raise_invalid_nonce(self, cairo_run, transaction):
+            private_key = generate_random_private_key()
+            address = int(private_key.public_key.to_checksum_address(), 16)
+            signed = Account.sign_transaction(transaction, private_key)
+            signature = [
+                *int_to_uint256(signed.r),
+                *int_to_uint256(signed.s),
+                signed["v"],
+            ]
             encoded_unsigned_tx = rlp_encode_signed_data(transaction)
+            tx_data = list(encoded_unsigned_tx)
 
-            with cairo_error(message="Invalid chain id"):
+            with cairo_error(message="Invalid nonce"), SyscallHandler.patch(
+                "Account_evm_address", address
+            ):
                 cairo_run(
-                    "test__validate",
-                    address=address,
-                    nonce=transaction["nonce"],
-                    chain_id=transaction["chainId"] + 1,
-                    r=int_to_uint256(signed.r),
-                    s=int_to_uint256(signed.s),
-                    v=signed["v"],
-                    tx_data=list(encoded_unsigned_tx),
+                    "test__execute_from_outside",
+                    tx_data=tx_data,
+                    signature=signature,
+                    chain_id=transaction.get("chainId") or CHAIN_ID,
                 )
 
+        @SyscallHandler.patch(
+            "IKakarot.get_cairo1_helpers_class_hash",
+            lambda addr, data: [CAIRO1_HELPERS_CLASS_HASH],
+        )
+        @SyscallHandler.patch("IKakarot.get_native_token", lambda addr, data: [0xDEAD])
+        @SyscallHandler.patch("IERC20.balanceOf", lambda addr, data: [0xDEAD, 0])
```

Comment:
this is not 0 but `Uint256(0xdead, 0)` or simple `0xdead`.
```suggestion
        @SyscallHandler.patch("IERC20.balanceOf", lambda addr, data: [0, 0])
```